src/vdom.rs

```diff
@@ -48,76 +153,114 @@ where
             )
             .expect("should append child OK");
 
-        // Diff and apply the `contents` against our dummy `<div/>`.
         let container = container.clone();
-        let mut vdom = Vdom {
-            component,
-            dom_buffers,
-            change_list,
-            container,
-            current_root,
-        };
-        vdom.render();
-        vdom
+        let inner = Rc::new(VdomInner {
+            shared: VdomInnerShared {
+                render_scheduled: Cell::new(None),
+            },
+            exclusive: RefCell::new(VdomInnerExclusive {
+                component: Some(component),
+                dom_buffers,
+                change_list,
+                container,
+                current_root,
+                events_registry: None,
+            }),
+        });
+
+        let (events_registry, events_trampoline) = EventsRegistry::new(Rc::downgrade(&inner));
+
+        {
+            let mut inner = inner.exclusive.borrow_mut();
+            inner.events_registry = Some(events_registry);
+            inner.change_list.init_events_trampoline(events_trampoline);
+
+            // Diff and apply the `contents` against our dummy `<div/>`.
+            inner.render();
+        }
+
+        Vdom { inner }
     }
 
-    /// Get a shared reference to the underlying render component.
+    /// Run this virtual DOM and its listeners forever and never unmount it.
     #[inline]
-    pub fn component(&self) -> &R {
-        &self.component
+    pub fn forget(self) {
+        mem::forget(self);
     }
 
-    /// Get an exclusive reference to the underlying render component.
+    /// Get a weak handle to this virtual DOM.
     #[inline]
-    pub fn component_mut(&mut self) -> &mut R {
-        &mut self.component
+    pub fn weak(&self) -> VdomWeak {
+        VdomWeak::new(&self.inner)
     }
 
-    /// Take the render component out of this virtual DOM.
+    /// Unmount this virtual DOM, unregister its event listeners, and return its
+    /// root render component.
     #[inline]
-    pub fn into_component(self) -> R {
-        self.component
+    pub fn unmount(self) -> Box<RootRender> {
+        Rc::try_unwrap(self.inner.clone())
+            .map_err(|_| ())
+            .unwrap_throw()
+            .exclusive
+            .into_inner()
+            .component
+            .take()
+            .unwrap_throw()
```

Comment:
Now this is a doozy of a chain of methods

Reply:
...yes